modules/service/replication-service.js

```diff
@@ -167,13 +185,15 @@ class ReplicationService {
 
         if (data) {
             this.logger.trace(`Loaded replication from cache for offer internal ID ${internalOfferId} and color ${color}`);
-            return data;
-        }
+        } else {
+            const offerDirPath = this._getOfferDirPath(internalOfferId);
+            const colorFilePath = path.join(offerDirPath, `${color}.json`);
 
-        const offerDirPath = this._getOfferDirPath(internalOfferId);
-        const colorFilePath = path.join(offerDirPath, `${color}.json`);
+            data = JSON.parse(await Utilities.fileContents(colorFilePath));
```

Comment:
if data is missing from cache, we should add it here, so we don't read from fs every time replication is requested

Reply:
We'll remove the replication cache to decrease memory usage during replications.